code/go/pkg/errors/processors/processor.go

```diff
@@ -0,0 +1,13 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package processors
+
+import "github.com/elastic/package-spec/v2/code/go/pkg/errors"
+
+// Processor is the interface for every lint processor.
+type Processor interface {
+	Process(errors.ValidationErrors) (errors.ValidationErrors, errors.ValidationErrors, error)
+	Name() string
+}
```

Comment:
Nit. Use names here for self-documentation.
```suggestion
type Processor interface {
	Process(errors.ValidationErrors) (processed errors.ValidationErrors, removed errors.ValidationErrors, error)
	Name() string
}
```

Or create an specific struct to return as we discussed at some point:
```suggestion
type Processor interface {
	Process(errors.ValidationErrors) (ProcessResult, error)
	Name() string
}

type ProcessResult struct {
	Processed errors.ValidationErrors
	Removed errors.ValidationErrors
}
```

I would use the `Processed`/`Removed` wording if we keep this as a general `Processor`. If we change it to something like a filter, we could talk about `Included`/`Excluded`.

Reply:
I'll go for the second option, creating an specific struct for these results :+1: